stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/Sprintf.java

```
@@ -95,19 +95,34 @@ public final void execute(final Context context) {
                 }
                 try {
                     char formatSpecifier = format.charAt(j);
+                    BRefType ref = args.getRefValue(k);
                     switch (formatSpecifier) {
                         case 'b':
                         case 'B':
                         case 'd':
                         case 'f':
-                            result.append(String.format("%" + padding + formatSpecifier, args.getRefValue(k).value()));
+                            if (ref != null) {
+                                result.append(String.format("%" + padding + formatSpecifier, ref.value()));
+                            } else {
+                                throw BLangExceptionHelper.getRuntimeException(
+                                        RuntimeErrors.ILLEGAL_FORMAT_CONVERSION,
+                                        format.charAt(j) + " != ()");
+                            }
                             break;
                         case 'x':
                         case 'X':
-                            formatHexString(args, result, k, padding, formatSpecifier);
+                            if (ref != null) {
+                                formatHexString(args, result, k, padding, formatSpecifier);
+                            } else {
+                                throw BLangExceptionHelper.getRuntimeException(
+                                        RuntimeErrors.ILLEGAL_FORMAT_CONVERSION,
+                                        format.charAt(j) + " != ()");
+                            }
```

Shall we invert the if condition, and fail fast?